### ISSUE TITLE
Tests 2056 and 2057 failing on macOS

### ISSUE THREAD

**Issue by jakirkham** [curl/curl#2394]
When building curl 7.58.0 on macOS 10.10 using Xcode 6.4, seeing tests 2056 and 2057 fail. Details can be found in [this build log]( https://travis-ci.org/conda-forge/curl-feedstock/builds/348677538#L4332-L4333 ) (linked to failing lines). Can confirm that these failures also occur on macOS 10.11 with Xcode 7.3.1. Also can confirm the issue persists in 7.59.0.

**Comment by jakirkham** [comment]
Unclear as to whether [this comment]( https://github.com/curl/curl/pull/1975#issuecomment-335436563 ) is relevant.

**Comment by jakirkham** [comment]
AFAIK this issue remains unresolved.

**Comment by bagder** [comment]
But nobody seems to work on it, why this is silent and the bot marked it as stale. Bugs that don't "move" will get closed eventually and will instead get added to the KNOWN_BUGS document.

**Comment by danielgustafsson** [comment]
Is it possible to get error logs and not just the FAILED` lines? I only have access to 10.13 but I can try and reproduce once I know what I'm looking for.

**Comment by jakirkham** [comment]
Ah! Was not aware there was a known bugs doc. Is that in this repo or somewhere else? Also would be great if the bot shared this info when posting as well.

The log is linked in the OP. It selects specific lines, but you are free to move around in the log as well as remove the selection. Please let me know if you have more questions.

**Comment by danielgustafsson** [comment]
The buildlog is there yes, but I would like to see the actual logfiles/output from the failing tests.

**Comment by bagder** [comment]
> would be great if the bot shared this info when posting as well.

It is mentioned in the "new issue" template. It was in that text you just skipped over =)

**Comment by jakirkham** [comment]
Hmm...I may have read it 6 months ago and forgot. :) In any event, would still be nice to have the bot mention it. ;)

**Comment by jakirkham** [comment]
@danielgustafsson, not sure I'm following. Can you please tell me how to get the additional information you need? We may need to try a new build to get at it. Happy to update the build to `cat` whatever files might be useful into the log.

**Comment by danielgustafsson** [comment]
I took a look at this today, and the reason why the tests are failing on macOS is that they depend on `LD_PRELOAD`, which doesn't work on macOS. There is actually code in `runtests.pl` to properly skip the tests in this case, but that codepath has a bug which doesn't exclude macOS from `LD_PRELOAD` and thus you are seeing this error. A fix proposed in #3106.

A longer term fix is to make this portable such that it uses `DYLD_INSERT_LIBRARIES` on macOS, but some quick testing shows that it's not as simple as the below diff so I'll add this to the TODO for now.
```
-LD_PRELOAD=%PWD/libtest/.libs/libstubgss.so
+DYLD_FORCE_FLAT_NAMESPACE=1
+DYLD_INSERT_LIBRARIES=%PWD/libtest/.libs/libstubgss.dylib
```
